scripts/SANS/sans/state/state_base.py

```diff
@@ -248,6 +254,18 @@ def _type_check(self, value):
                                                                                str(value), type(value)))
 
 
+class PositiveFloatListParameter(TypedParameter):
+    def __init__(self):
+        super(PositiveFloatListParameter, self).__init__(list, all_list_elements_are_float_and_positive_and_not_empty)
+
+    def _type_check(self, value):
+
+        if not isinstance(value, self.parameter_type) or not all_list_elements_are_float_and_not_empty(value):
+            raise TypeError("Trying to set {0} which expects a value of type {1}."
+                            " Got a value of {2} which is of type: {3}".format(self.name, str(self.parameter_type),
+                                                                               str(value), type(value)))
```

Comment:
Are the `str()` conversions needed here? Wouldn't `format()` do this automatically?